src/queries/label-text.js

```diff
@@ -51,21 +52,25 @@ function queryAllByLabelText(
         // <label for="someId">text</label><input id="someId" />
 
         // .control support has landed in jsdom (https://github.com/jsdom/jsdom/issues/2175)
-        return container.querySelector(`[id="${label.getAttribute('for')}"]`)
+        elementsForLabel.push(
+          container.querySelector(`[id="${label.getAttribute('for')}"]`),
+        )
       }
       if (label.getAttribute('id')) {
         // <label id="someId">text</label><input aria-labelledby="someId" />
-        return container.querySelector(
-          `[aria-labelledby~="${label.getAttribute('id')}"]`,
-        )
+        container
+          .querySelectorAll(`[aria-labelledby~="${label.getAttribute('id')}"]`)
+          .forEach(element => elementsForLabel.push(element))
       }
       if (label.childNodes.length) {
         // <label>text: <input /></label>
-        return label.querySelector(selector)
+        label
+          .querySelectorAll('input, textarea')
```

Comment:
This is insufficient. We need to select *every* form control element type. Unfortunately there's not a common class that all form control elements inherit from, but I think we'd be safe if we found every element that supports [the `.labels` property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/labels).

I'm not sure of the best way to do this, but here's a list of all the element types: https://developer.mozilla.org/en-US/docs/Web/API (Everything named something like HTML*Element).

Reply:
I'll look into that and see if I can find all of them. :+1: